docs/guides/workflow.md

```diff
@@ -0,0 +1,16 @@
+# Workflow
```

Comment:
Feels like an introduction?

Reply:
It's introductory in the sense that it teaches people specifically about where it fits into the grand scheme of things, which you might not need to be told if you are already very aware of the API design-first workflow. I put it in as a guide for people who are not.